### PR TITLE
Developer Guide. Use Case Diagram, Sequence Diagrams and Code quality !!

### PR DESCRIPTION
**Pull request by MarcusGitty**
Hi,

I've added the sequence diagrams for my view stats command and Use case diagram for tuteetally.

I've also fixed one of the ghost of "StatsCommand" sequence diagram

### CODE REVIEW

File: docs/DeveloperGuide.md [hunk 1]
```diff
@@ -205,7 +206,7 @@ If more than one valid prefix `-all`, `-name`, `-id`, or `stats` are present, `V
 
 Below is an example usage scenario where the `view -stats` command was entered.
 
-**Step 1:** User first calls `view -stats`. The input is passed into ``AddressBookParser`` which instantiates a ``ViewCommandParser`` instance.
+**Step 1:** User first calls `view -stats`. The input is passed into ``AddressBookParser`` which instantiates a ``1CommandParser`` instance.
```
Comment (0nesheep) [inline docs/DeveloperGuide.md:209]: Hmm not sure if this is a typo but what does `1CommandParser` refer to?

File: docs/DeveloperGuide.md [hunk 2]
```diff
@@ -245,13 +246,61 @@ Below is a sequence diagram of how view all interacts with multiple classes.
...
+The frame below shows how the logic class gets the command result.
+
+<puml src="diagrams/ViewStatsSequenceDiagramGetCommandResult.puml" alt="ViewStatsSequenceDiagramGetCommandResult" />
+
+The CommandResult will then be returned to the UIManager and a SummaryStatsWindow Instance will be created, the Sequence diagram below shows how
+it get the SummaryStats from `Logic`the respective frame will show `SummaryStatsWindow::updateSummaryStats` clearly. 
+
+<puml src="diagrams/ViewStatsSequenceDiagramUpdateTotalCountofPersons.puml" alt="ViewStatsSequenceDiagramUpdateTotalCountofPerson" />
+<puml src="diagrams/ViewStatsSequenceDiagramUpdateTotalOwingsOfPerson.puml" alt="ViewStatsSequenceDiagramUpdateTotalOwingsOfPerson" />
+<puml src="diagrams/ViewStatsSequenceDiagramUpdateUpcomingExams.puml" alt="ViewStatsSequenceDiagramUpdateUpcomingExams" />
```
Comment (0nesheep) [inline docs/DeveloperGuide.md:278]: woaaa nice!

**Review by 0nesheep — APPROVED** [review]
LGTM!

**Review by 0nesheep — CHANGES_REQUESTED** [review]
Sorry oops! Supposed to request for changes for the typo in developer guide line 209

File: docs/DeveloperGuide.md (hunk 1 above)
Comment (MarcusGitty) [reply docs/DeveloperGuide.md:209]: Oh my yes it's a typo xD

**Review by 0nesheep — APPROVED** [review]
LGTM! Nice reformatting of strings and removal of redundant test cases!

**Pull request merged**